src/lib/lifeCycle.ts

```diff
@@ -61,13 +62,49 @@ import {
   recursivelyFindProjectRootDir,
   setProjectRootDir,
 } from "./paths.js";
-import { confirm } from "./prompt.js";
+import { confirm, list } from "./prompt.js";
+
+const NO_DOCS_IN_CONFIGS =
+  "Only include a comment with a link to the online docs";
+
+const DOCS_IN_CONFIGS = "Include commented-out examples and docs";
+
+const DOCS_IN_CONFIGS_OPTIONS = [NO_DOCS_IN_CONFIGS, DOCS_IN_CONFIGS] as const;
+
+const handleDocsInConfigs = async (userConfig: UserConfig) => {
+  const docsInConfigsOption = isInteractive
+    ? await list({
+        message:
+          "Select what comments you want to see in the configs generated by the CLI",
+        oneChoiceMessage: () => {
+          return commandObj.error("Unreachable");
+        },
+        onNoChoices: () => {
+          return commandObj.error("Unreachable");
+        },
+        options: [...DOCS_IN_CONFIGS_OPTIONS],
+      })
+    : NO_DOCS_IN_CONFIGS;
+
+  userConfig.docsInConfigs = docsInConfigsOption === DOCS_IN_CONFIGS;
+  await userConfig.$commit();
+
+  commandObj.logToStderr(
+    `If you change your mind later, modify "docsInConfigs" property in ${userConfig.$getPath()}`,
+  );
+};
 
 const ensureUserConfig = async (): Promise<void> => {
   const maybeUserConfig = await initUserConfig();
 
   if (maybeUserConfig !== null) {
-    setUserConfig(maybeUserConfig);
+    const userConfig = maybeUserConfig;
+
+    if (userConfig.docsInConfigs === undefined) {
+      await handleDocsInConfigs(userConfig);
+    }
+
+    setUserConfig(userConfig);
```

Comment:
How `handleDocsInConfigs` is different from `setUserConfig`?

this code looks a little weird: you call `handleDocsInConfigs` and `setUserConfig` twice in the function. One time inside `if`, another time at the end.

Can you make it so it's called only a single time? So it reads easier, right now it's really hard to tell what will happen with all these IFs

Reply:
1. These are two completely different things. One "handles" docsInConfigs property in the configs - that is to say it determines weather it should be true or false. The other is setting this config as a global singelton (because it's very convenient to use and you can only have onle user config)
2. I can't do change so it is called only once because I am handlinngle two completely different possbilities: one is when there is no config at all. The other one is when there is a config but this property is missing in it (all our existing users fall into the latter cattegory)
3. I will try to improve the code